### PR TITLE
Switch Cro::HTTP::Client to parse IRIs

### PR DESCRIPTION
**Pull request by vendethiel**
This depends on croservices/cro-core#28

### CODE REVIEW

File: lib/Cro/Uri/HTTP.pm6 [hunk 1]
```diff
@@ -1,43 +1,10 @@
 use Cro::Uri :decode-percents, :encode-percents;
 use Cro::HTTP::MultiValue;
 
-class Cro::Uri::HTTP is Cro::Uri {
```
Comment (vendethiel) [inline lib/Cro/Uri/HTTP.pm6:4]: That name wants bikeshedding + wants to be in another file

File: lib/Cro/HTTP/Request.pm6 [hunk 2]
```diff
@@ -5,7 +5,7 @@ use Cro::HTTP::BodyParserSelectors;
 use Cro::HTTP::BodySerializerSelectors;
 use Cro::HTTP::Message;
 use Cro::TLS;
-use Cro::Uri::HTTP;
+use Cro::Iri::HTTP;
```
Comment (jnthn) [inline lib/Cro/HTTP/Request.pm6:8]: Given this is the HTTP request object the HTTP protocol rules apply, so I think this should probably be URI.

File: lib/Cro/Uri/HTTP.pm6 (hunk 1 above)
Comment (jnthn) [reply lib/Cro/Uri/HTTP.pm6:4]: Maybe `Cro::ResourceIdentifier::HTTP`.

**Comment by Altai-man** [comment]
Merging in a dev branch to get a write access.

**Pull request merged**